subsys: uart_async_adapter: Provide UART Async Adapter (#14490)
nrfconnect/sdk-nrf: samples/bluetooth/peripheral_uart/README.rst

Async adapter experimental module
---------------------------------

The default sample configuration uses the UART async API.

> To communicate with USB CDC ACM the :ref:`lib_uart_async_adapter_` module is used.

Reviewer: What is the library you want to link to?

> To communicate with USB CDC ACM the :ref:`lib_uart_async_adapter_` module is used.

Reviewer: ```suggestion
To communicate with USB CDC ACM the :ref:`lib_uart_async_adapter` module is used.
```
Ah, you wanted to link to the new library you are writing.

> The sample uses the :ref:`lib_uart_async_adapter` library to communicate with USB CDC ACM.

Reviewer: ```suggestion
The sample uses the :ref:`lib_uart_async_adapter` library to communicate with the USB CDC ACM driver.
```

This is needed because the USB CDC ACM implementation provides only the interrupt interface.

> The module requires the :kconfig:option:`CONFIG_UART_ASYNC_ADAPTER` to be set to ``y``.

Reviewer: ```suggestion
To use the library, set the :kconfig:option:`CONFIG_UART_ASYNC_ADAPTER` Kconfig option to ``y``.
```

> For more information about the adapter, see the :ref:`lib_uart_async_adapter_` documentation.

Reviewer: ```suggestion
For more information about the adapter, see the :ref:`lib_uart_async_adapter` documentation.
```

> For more information about the adapter, see the :ref:`lib_uart_async_adapter` documentation.

Reviewer: ```suggestion
```
You already linked to the library doc above, so not needed here.
